tests/lib/compose-features.py

```diff
@@ -0,0 +1,235 @@
+                                     file for each <backend>:<system> with feature-tagging information, accompanied with
+                                     additional test information.
+
+                                     Composed feature consolidation mode
+
+                                     Given a directory containing files of pre-composed feature information with filenames like
+                                     <system>_<run-attempt>.json, it writes the consolidated feature information in a new
+                                     directory (specified with the --output flag) where the latest rerun data replaces the old.
+                                     So if a file contains one test that was later rerun, the new consolidated file will contain
+                                     unaltered content from the original run except for the one test rerun that will replace
+                                     the old.
+                                     """, formatter_class=argparse.RawDescriptionHelpFormatter)
+    parser.add_argument('-d', '--dir', type=str, required=True,
+                        help='Path to the folder containing json files')
+    parser.add_argument('-o', '--output', type=str,
+                        help='Output directory', required=True)
+    parser.add_argument('-s', '--scenarios', type=str,
+                        help='Comma-separated list of scenarios', default="")
+    parser.add_argument('-e', '--env-variables', type=str,
+                        help='Comma-separated list of environment variables as key=value', default="")
+    parser.add_argument('-f', '--failed-tests', type=str,
+                        help='List of failed tests', default="")
+    parser.add_argument('--run-attempt', type=run_attempt_type, help="""Run attempt number of the json files contained in the folder [1,). 
+                        Only needed when rerunning spread for failed tests. When specified, will append the run attempt 
+                        number on the filename, which will then be used when running this script with the --replace-old-runs
+                        flag to determine replacement order""")
+    parser.add_argument('-r', '--replace-old-runs', action="store_true",
+                        help='When set, will process pre-composed runs and consolidate them into the output dir')
+    args = parser.parse_args()
+
+    if args.replace_old_runs:
+        replace_old_runs(args.dir, args.output)
+        exit(0)
+
+    attempt = ""
+    if args.run_attempt:
+        attempt = "_%s" % args.run_attempt
+    os.makedirs(args.output, exist_ok=True)
+    systems = get_system_list(args.dir)
+    for system in systems:
+        composed = compose_system(dir=args.dir, system=system,
+                                  failed_tests=args.failed_tests, env_variables=args.env_variables)
+        system = "_".join(system.split(':'))
```

Comment:
You can just replace:
```suggestion
        system = system.replace(":", "_")
```